demo/fake-api.js

```diff
@@ -0,0 +1,595 @@
+          },
+          PasswordsInArguments: {
+            score: 0.0,
+            name: "Passwords in Java Arguments",
+            topic: "jvm_information",
+            description:
+              "Summary:\nThe recording does not seem to contain passwords in the application arguments.",
+          },
+        }
+      );
+    });
+    this.get("/api/v1/targets/:targetId/recordingOptions", () => []);
+    this.get("/api/v1/targets/:targetId/events", () => [
+      {
+        category: ["GC", "Java Virtual Machine"],
+        name: "GC Heap Configuration",
+        typeId: "jdk.GCHeapConfiguration",
+        description: "The configuration of the garbage collected heap",
+      },
+    ]);
+    this.get("/api/v1/targets/:targetId/templates", () => [
+      {
+        name: "Demo Template",
+        provider: "Demo",
+        type: "TARGET",
+        description: "This is not a real event template, but it is here!",
+      },
+    ]);
+    this.get("/api/v2/probes", () => []);
+
+    this.post("/api/v2/rules", (schema, request) => {
+      let attrs = JSON.parse(request.requestBody);
+      return {
+        data: {
+          result: schema.rules.create(attrs),
+        },
+      };
+    });
+    this.get("/api/v2/rules", (schema) => ({
+      data: { result: schema.rules.all().models },
+    }));
+    this.get("/api/v2.2/credentials", () => ({ data: { result: [] } }));
+
+    this.post("/api/v2.2/graphql", (schema, request) => {
+      var body = JSON.parse(request.requestBody);
+      var query = body.query.trim();
+      var variables = body.variables;
+      var begin = query.substring(0, query.indexOf("{"));
+      var name = "unknown";
+      for (var n of begin.split(" ")) {
+        if (n == "{") {
+          break;
+        }
+        if (!n || n == "query") {
+          continue;
+        }
+        name = n.substring(0, n.indexOf("("));
+        break;
+      }
+      if (name === "unknown" || !name) {
+        return new Response(
+          400,
+          {},
+          `${JSON.stringify(
+            request.url
+          )} (query: '${name}') currently unsupported in demo`
+        );
+      }
+      let data = {};
+      switch (name) {
+        case "ArchivedRecordingsForTarget":
+        case "UploadedRecordings":
+          data = {
+            archivedRecordings: {
+              data: schema.archives.all().models,
+            },
+          };
+          break;
+        case "ActiveRecordingsForTarget":
+          data = {
+            targetNodes: [
+              {
+                recordings: {
+                  archived: {
+                    data: schema.archives.all().models,
```

Comment:
Should be `schema.recordings.all().models`?

Reply:
Maybe? This was just what I observed in reverse engineering the Recordings view based on actual network requests being made.

1. Create a new Active recording
2. Wait a few moments, then archive it
3. Switch to the Archived Recordings tab
4. Select the recording and Edit Labels

If you do this against a live Cryostat instance with your browser network inspector open, you'll see this GraphQL `"query ActiveRecordingsForTarget($connectUrl: String) { targetNodes(filter: { name: $connectUrl }) { recordings { archived { data { name downloadUrl reportUrl metadata { labels } } } } } }"` request. The server response looks like this:

```json
{
	"data": {
		"targetNodes": [
			{
				"recordings": {
					"archived": {
						"data": [
							{
								"name": "cryostat_foo_20230509T170101Z.jfr",
								"downloadUrl": "https://localhost:8181/api/beta/recordings/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:0%2Fjmxrmi/cryostat_foo_20230509T170101Z.jfr",
								"reportUrl": "https://localhost:8181/api/beta/reports/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:0%2Fjmxrmi/cryostat_foo_20230509T170101Z.jfr",
								"metadata": {
									"labels": {
										"template.name": "Profiling",
										"template.type": "TARGET"
									}
								}
							}
						]
					}
				}
			}
		]
	}
}
```

I'm not sure if this is a server bug, or perhaps more likely, the web-client just has the wrong query name for what it's really asking for here.

Reply:
Oh yeh I think its query name that is being mistyped.

https://github.com/cryostatio/cryostat-web/blob/0213921fb1fa686a7c42e8b6fd24df42ab0eb4be/src/app/RecordingMetadata/BulkEditLabels.tsx#L206

We will need to fix that on the web-client.